packages/web/src/generators/static-serve/static-serve-configuration.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  convertNxGenerator,
+  logger,
+  parseTargetString,
+  readProjectConfiguration,
+  stripIndents,
+  TargetConfiguration,
+  Tree,
+  updateProjectConfiguration,
+} from '@nrwl/devkit';
+import { Schema as FileServerExecutorSchema } from '../../executors/file-server/schema.d';
+interface WebStaticServeSchema {
+  buildTarget: string;
+  outputPath?: string;
+  targetName?: string;
+}
+
+interface NormalizedWebStaticServeSchema extends WebStaticServeSchema {
+  projectName: string;
+  targetName: string;
+}
+
+export function webStaticServeGenerator(
+  tree: Tree,
+  options: WebStaticServeSchema
+) {
+  const opts = normalizeOptions(tree, options);
+  addStaticConfig(tree, opts);
+}
+
+function normalizeOptions(
+  tree: Tree,
+  options: WebStaticServeSchema
+): NormalizedWebStaticServeSchema {
+  const target = parseTargetString(options.buildTarget);
+  const opts: NormalizedWebStaticServeSchema = {
+    ...options,
+    targetName: options.targetName || 'serve-static',
+    projectName: target.project,
+  };
+
+  const projectConfig = readProjectConfiguration(tree, target.project);
+  const buildTargetConfig = projectConfig?.targets?.[target.target];
+  if (!buildTargetConfig) {
+    throw new Error(stripIndents`Unable to read the target configuration for the provided build target, ${opts.buildTarget}
+Are you sure this target exists?`);
+  }
+
+  if (projectConfig.targets[opts.targetName]) {
+    throw new Error(stripIndents`Project ${target.project} already has a '${opts.targetName}' target configured.
+Either rename or remove the existing '${opts.targetName}' target and try again.
+Optionally, you can provide a different name with the --target-name option other than '${opts.targetName}'`);
+  }
+
+  // NOTE: @nrwl/web:file-server only looks for the outputPath option
+  if (!buildTargetConfig.options?.outputPath && !opts.outputPath) {
+    // attempt to find the suiteable path from the outputs
+    let maybeOutputValue: any;
+    for (const o of buildTargetConfig?.outputs || []) {
+      const isInterpolatedOutput = o.trim().startsWith('{options.');
+      if (!isInterpolatedOutput) {
+        continue;
+      }
+      const noBracketParts = o.replace(/[{}]/g, '').split('.');
+
+      if (noBracketParts.length === 2 && noBracketParts?.[1]) {
+        const key = noBracketParts[1].trim();
+        const value = buildTargetConfig.options?.[key];
+        if (value) {
+          maybeOutputValue = value;
+          break;
+        }
+      }
+    }
```

Comment:
unsure if we want to do this, but figured trying to find the output if the target doesn't have 'outputPath' would be helpful for those setting up the static target